test-util/src/test/java/care/smith/fts/test/FhirGeneratorTest.java

```diff
@@ -1,22 +1,22 @@
 package care.smith.fts.test;
 
+import static care.smith.fts.util.FhirUtils.toBundle;
+import static java.util.UUID.randomUUID;
 import static org.assertj.core.api.Assertions.assertThat;
 
 import care.smith.fts.util.FhirUtils;
 import java.io.IOException;
+import java.util.stream.Stream;
 import org.hl7.fhir.r4.model.Bundle;
 import org.junit.jupiter.api.Test;
 
 class FhirGeneratorTest {
 
   @Test
   void serializeAndDeserializeGenerateBundle() throws IOException {
-    FhirGenerator gicsConsentGenerator = new FhirGenerator("GicsResponseTemplate.json");
-    gicsConsentGenerator.replaceTemplateFieldWith(
-        "$QUESTIONNAIRE_RESPONSE_ID", new FhirGenerator.UUID());
-    gicsConsentGenerator.replaceTemplateFieldWith("$PATIENT_ID", new FhirGenerator.UUID());
-
-    Bundle bundle = gicsConsentGenerator.generateBundle(1, 200);
+    var gicsConsentGenerator =
+        FhirGenerators.gicsResponse(() -> randomUUID().toString(), () -> randomUUID().toString());
```

Comment:
Looking at this line of code, one could consider using a builder pattern explicitly (but optionally) set patientId and QuestionaireResponseId, making it clearer for the reader.